install/ui/src/freeipa/vault.js

```diff
@@ -0,0 +1,818 @@
+            },
+            'cn',
+            'description',
+            {
+                $type: 'radio',
+                name: 'ipavaulttype',
+                default_value: 'standard',
+                read_only: true,
+                options: [
+                    {
+                        value: 'standard',
+                        label: '@i18n:objects.vault.standard_type'
+                    },
+                    {
+                        label: '@i18n:objects.vault.symmetric_type'
+                    },
+                    {
+                        label: '@i18n:objects.vault.asymmetric_type'
+                    }
+                ],
+                tooltip: "@i18n:objects.vault.type_tooltip"
+            }
+        ]
+    };
+
+        var section_warn_standard = {
+            name: 'warning_st',
+            fields: [
+                {
+                    field: false,
+                    $type: 'html',
+                    name: 'warn_standard'
+                }
+            ],
+            layout: {
+                $factory: widget.fluid_layout,
+                widget_cls: "col-sm-12 controls",
+                label_cls: "hide"
+            }
+        };
+
+    spec.sections.push(section_warn_arch_ret);
+    spec.sections.push(section_f);
+    spec.sections.push(section_warn_standard);
+
+    var that = IPA.entity_adder_dialog(spec);
+
+    that.create_add_command = function(record) {
+        var command = that.entity_adder_dialog_create_add_command(record);
+
+        var type_f = that.fields.get_field('type');
+        var type = type_f.save()[0];
+
+        if (type === 'shared') command.set_option(type, true);
+
+        return command;
+    };
+
+    that.create_content = function() {
+        var warn_st_w = that.widgets.get_widget('warning_st.warn_standard');
+        var warn_arch_ret_w = that.widgets.get_widget('warning_ar.warn_arch_ret');
+        var warn_standard = $('<div />', {
+            'class': 'alert alert-info',
+            text: text.get("@i18n:objects.vault.add_warn_standard")
+        });
+
+        $('<i />', {
+            'class': 'fa fa-info-circle'
+        }).appendTo(warn_standard);
+
+        warn_st_w.html = warn_standard;
+
+        var warn_arch_ret = $('<div />', {
+            'class': 'alert alert-info',
+            text: text.get("@i18n:objects.vault.add_warn_arch_ret")
+        });
+
+        $('<i />', {
+            'class': 'fa fa-info-circle'
+        }).appendTo(warn_arch_ret);
+
+        warn_arch_ret_w.html = warn_arch_ret;
+
+        that.entity_adder_dialog_create_content();
+
+        var facet_name = that.entity.facet.name;
+        facet_name = facet_name.substr(0, facet_name.indexOf('_'));
+
+        that.fields.get_field('type').set_pristine_value([facet_name]);
+    };
+
+    return that;
+};
+
+
+/**
+ * Creates specification of search facet for User Vaults
+ */
+var make_user_vault_search_spec = function() {
+    return {
```

Comment:
In self-service, user cannot add vault because 'add' button is hidden. He can add stadard vault on CLI though.

It is hidden because of `spec.hide_cond = spec.hide_cond || ['self-service'];` in `IPA.add_action `